java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/TimeConsumer.java

```diff
@@ -42,56 +42,26 @@ public static TimeConsumer createConsumer(
     }
   }
 
-  protected TimeMilliVector vector;
-  protected final int columnIndexInResultSet;
-  protected final Calendar calendar;
-
-  protected int currentIndex;
-
-  /**
-   * Instantiate a TimeConsumer.
-   */
-  public TimeConsumer(TimeMilliVector vector, int index) {
-    this(vector, index, null);
-  }
-
-  /**
-   * Instantiate a TimeConsumer.
-   */
-  public TimeConsumer(TimeMilliVector vector, int index, Calendar calendar) {
-    this.vector = vector;
-    this.columnIndexInResultSet = index;
-    this.calendar = calendar;
-  }
-
-  @Override
-  public void close() throws Exception {
-    this.vector.close();
-  }
-
-  @Override
-  public void resetValueVector(TimeMilliVector vector) {
-    this.vector = vector;
-    this.currentIndex = 0;
-  }
-
   /**
    * Nullable consumer for {@link TimeMilliVector}.
    */
-  static class NullableTimeConsumer extends TimeConsumer {
+  static class NullableTimeConsumer extends BaseConsumer<TimeMilliVector> {
+
+    protected final Calendar calendar;
 
     /**
      * Instantiate a TimeConsumer.
      */
     public NullableTimeConsumer(TimeMilliVector vector, int index) {
-      super(vector, index);
+      this(vector, index, null);
```

Comment:
/*calendar=*/null for last parameter please.

Reply:
Good point. Thanks.